### PR TITLE
Progress on UDF documentation for `pixeltable.functions` and `pixeltable.ext.functions` modules

### PR DESCRIPTION
**Pull request by aaron-siegel**

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/233)

File: pixeltable/ext/functions/yolox.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ def yolox(images: Batch[PIL.Image.Image], *, model_id: str, threshold: float = 0
 
 @pxt.udf
 def yolo_to_coco(detections: dict) -> list:
+    """Converts YOLOX detection results to COCO format."""
```
Comment (mkornacker) [inline pixeltable/ext/functions/yolox.py]: Is this really specific to YOLOX, or does it apply to all YOLO variants?

File: pixeltable/functions/fireworks.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ def chat_completions(
     top_p: Optional[float] = None,
     temperature: Optional[float] = None,
 ) -> dict:
+    """Pixeltable UDF for the Fireworks `chat/completions` API endpoint."""
```
Comment (mkornacker) [inline pixeltable/functions/fireworks.py]: No need to preface with 'Pixeltable UDF for ...', they all are.

(The docstring for str.capitalize() doesn't start with "Python function for ...")

File: pixeltable/functions/huggingface.py [hunk 3]
```diff
@@ -129,7 +129,27 @@ def cross_encoder_list(sentence1: str, sentences2: list, *, model_id: str) -> li
 
 @pxt.udf(batch_size=32, return_type=ts.ArrayType((None,), dtype=ts.FloatType(), nullable=False))
 def clip_text(text: Batch[str], *, model_id: str) -> Batch[np.ndarray]:
-    """Runs the specified CLIP model on text."""
+    """
+    Runs the specified CLIP embedding model on text. `model_id` should be a reference to a pretrained
```
Comment (mkornacker) [inline pixeltable/functions/huggingface.py]: "Runs the ..." also sounds weird.

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/233)* status: 0 of 16 files reviewed, 3 unresolved discussions (waiting on @mkornacker)

___
*[`pixeltable/ext/functions/yolox.py` line 61 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/233#gh-1697730283:-O3cs9pD8edeq54QH5TF:b26p524) ([raw file](https://github.com/pixeltable/pixeltable/blob/49f6823faa37da8ff6c9bb286f58d6c1fb6ebe86/pixeltable/ext/functions/yolox.py#L61)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Is this really specific to YOLOX, or does it apply to all YOLO variants?
</blockquote></details>

I don't know. I don't know if we should be making promises about it without researching the other major yolo variants first?
___
*[`pixeltable/functions/fireworks.py` line 32 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/233#gh-1697730682:-O3csNEx5vqtP5oH24g7:bt2vr1b) ([raw file](https://github.com/pixeltable/pixeltable/blob/49f6823faa37da8ff6c9bb286f58d6c1fb6ebe86/pixeltable/functions/fireworks.py#L32)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

No need to preface with 'Pixeltable UDF for ...', they all are.

(The docstring for str.capitalize() doesn't start with "Python function for ...")
</blockquote></details>

Fixed.
___
*[`pixeltable/functions/huggingface.py` line 133 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/233#gh-1697730927:-O3fGknx9AzkAFupic7d:bt2vr1b) ([raw file](https://github.com/pixeltable/pixeltable/blob/49f6823faa37da8ff6c9bb286f58d6c1fb6ebe86/pixeltable/functions/huggingface.py#L133)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

"Runs the ..." also sounds weird.
</blockquote></details>

Fixed.

File: mkdocs.yml [hunk 4]
```diff
@@ -29,12 +29,16 @@ nav:
       - FrameIterator: api/iterators/frame-iterator.md
     - DataFrame: api/data-frame.md
     - Table: api/table.md
-  - Pixeltable Functions:
-    - Huggingface: api/functions/huggingface.md
+  - Built-in UDFs:
```
Comment (mkornacker) [inline mkdocs.yml]: It's a bit weird to call this "built-in user-defined functions" (they're built-in, not user-defined).

We'll also have to incorporate operators into this section, similar to how Postgres structures their docs.

File: mkdocs.yml [hunk 5]
```diff
@@ -29,12 +29,16 @@ nav:
...
     - image: api/functions/image.md
     - string: api/functions/string.md
     - timestamp: api/functions/timestamp.md
     - video: api/functions/video.md
+    - fireworks: api/functions/fireworks.md
```
Comment (mkornacker) [inline mkdocs.yml:38]: I feel like these are thematically different than the functions for our supported data types.

File: pixeltable/ext/functions/yolox.py [hunk 6]
```diff
@@ -22,14 +22,28 @@
 @pxt.udf(batch_size=4)
 def yolox(images: Batch[PIL.Image.Image], *, model_id: str, threshold: float = 0.5) -> Batch[dict]:
     """
-    Runs the specified YOLOX object detection model on an image.
+    Computes YOLOX object detections for the specified image. `model_id` should reference one of the models
+    defined in the [YOLOX documentation](https://github.com/Megvii-BaseDetection/YOLOX).
 
     YOLOX support is part of the `pixeltable.ext` package: long-term support is not guaranteed, and it is not
```
Comment (mkornacker) [inline pixeltable/ext/functions/yolox.py:28]: Maybe we should have an 'ext' section in the docs? I expect that to get more voluminous.

**Review by mkornacker — APPROVED** [review]

**Pull request merged**